graspy/inference/latent_distribution_test.py

```diff
@@ -14,121 +14,207 @@
     dot product graphs have the same distributions of latent positions.
-    
+
     This test can operate on two graphs where there is no known matching between
-    the vertices of the two graphs, or even when the number of vertices is different. 
+    the vertices of the two graphs, or even when the number of vertices is different.
     Currently, testing is only supported for undirected graphs.
 
     Read more in the :ref:`tutorials <inference_tutorials>`
 
     Parameters
     ----------
+    test : str
+        Independence test to use, one of ["CCA", "Dcorr", "HHG", "RV", "Hsic", "MGC"].
+
+    metric : str or function, (default="euclidean")
+        Distance metric to use, either a callable or a valid string.
+        The callable should behave similarly to sklearn.metrics.pairwise_distances.
+        Valid strings: https://scikit-learn.org/stable/modules/generated/sklearn.metrics.pairwise_distances.html.
+
     n_components : int or None, optional (default=None)
         Number of embedding dimensions. If None, the optimal embedding
         dimensions are found by the Zhu and Godsi algorithm.
 
     n_bootstraps : int (default=200)
         Number of bootstrap iterations.
 
-    bandwidth : float, optional (default=0.5)
-        Bandwidth to use for gaussian kernel. If None,
-        the median heuristic will be used.
+    num_workers : int, optional (default=1)
+        Number of workers to use. If more than 1, parallelizes the code.
 
     Attributes
     ----------
     sample_T_statistic_ : float
-        The observed difference between the embedded latent positions of the two 
+        The observed difference between the embedded latent positions of the two
         input graphs.
 
-    p_ : float
+    p_value_ : float
         The overall p value from the test.
-    
-    null_distribution_ : ndarray, shape (n_bootstraps, )
+
+    null_distribution_ : None or ndarray, shape (n_bootstraps, )
         The distribution of T statistics generated under the null.
 
     References
     ----------
-    .. [1] Tang, M., Athreya, A., Sussman, D. L., Lyzinski, V., & Priebe, C. E. (2017). 
+    .. [1] Tang, M., Athreya, A., Sussman, D. L., Lyzinski, V., & Priebe, C. E. (2017).
         "A nonparametric two-sample hypothesis testing problem for random graphs."
         Bernoulli, 23(3), 1599-1630.
+
+    .. [2] Panda, S., Palaniappan, S., Xiong, J., Bridgeford, E., Mehta, R., Shen, C., & Vogelstein, J. (2019).
+        "hyppo: A Comprehensive Multivariate Hypothesis Testing Python Package."
+        arXiv:1907.02088.
     """
 
-    def __init__(self, n_components=None, n_bootstraps=200, bandwidth=None):
+    def __init__(
+        self,
+        test,
+        metric="euclidean",
+        n_components=None,
+        n_bootstraps=200,
+        num_workers=1,
+    ):
+
+        _VALID_TESTS = ["cca", "dcorr", "hhg", "rv", "hsic", "mgc"]
+        _VALID_METRICS = [
```

Comment:
good point, I can do code slightly differently to avoid this

Reply:
still seems weird to have here

Reply:
@bdpedigo do we need to have any metrics with keywords other than euclidian and gaussian? how common is it to use any other one? if it's not that common, maybe all other ones should be provided as a function object by practitioner, rather than us having a million of options as strings?

Reply:
i think this has been paired down a lot to just the ones in sklearn PAIRED_DISTANCES.keys() whiich are basically cosine, euclidean, manhattan